packages/supabase_flutter/lib/src/supabase_auth.dart

```diff
@@ -58,24 +58,26 @@ class SupabaseAuth with WidgetsBindingObserver {
       final persistedSession = await _localStorage.accessToken();
       if (persistedSession != null) {
         try {
-          Supabase.instance.client.auth.setInitialSession(persistedSession);
+          await Supabase.instance.client.auth
+              .setInitialSession(persistedSession);
+          shouldEmitInitialSession = false;
```

Comment:
For some reason the `shouldEmitInitialSession` was never set to false, causing the initialSession to be emitted every single time.